### ISSUE TITLE
zraxx - Function _pendingPoints cannot return correct value

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2024-05-sophon-judging#69]
zraxx

medium

# Function _pendingPoints cannot return correct value

## Summary

Function _pendingPoints cannot return correct value

## Vulnerability Detail

The function `_pendingPoints` is used to calculate the number of points to be accumulated by the user. However, its points are calculated slightly differently with `updatePool`. In function `_pendingPoints`, `accPointsPerShare` is multiplied by 1e18, while in function `updatePool`, it dosen’t. This subtle difference will cause the value calculated by the function `_pendingPoints` to be different from the actual value.

For example, when initial pool.accPointsPerShare = 0, blockMultiplier = 2e18, pointsPerBlock = 100, pool.allocPoint = 9, totalAllocPoint = 100, pool.amount = 10e18, user.amount = 5e18.

So, in the function `_pendingPoints` , accPointsPerShare = 2e18 * 100 * 9 / 100 * 1e18 / 10e18 = 1.8e18, addPoints = user.amount * accPointsPerShare / 1e36 = 1.8e18 * 5e18 / 1e36 = 9

While in the function `updatePool` , accPointsPerShare = 2e18 * 100 * 9 / 100 / 10e18 = 1, addPoints = user.amount * accPointsPerShare / 1e18 = 1 * 5e18 / 1e18 = 5.

In other words, the deviation reaches 80% ( (9-5) / 5 ).

## Impact

User will get a wrong pendingPoints.

## Code Snippet

https://github.com/sherlock-audit/2024-05-sophon/blob/main/farming-contracts/contracts/farm/SophonFarming.sol#L357-L384

## Tool used

Manual Review

## Recommendation

Don't multiply accPointsPerShare with 1e18.

**Comment by sherlock-admin2** [comment]
1 comment(s) were left on this issue during the judging contest.

**0xmystery** commented:
>  invalid because getPendingPoints() will not practically be used till point farming has ended. The higher precision adopted is by design
